.palantir/revapi.yml

```diff
@@ -877,6 +877,10 @@ acceptedBreaks:
     - code: "java.field.serialVersionUIDChanged"
       new: "field org.apache.iceberg.util.SerializableMap<K, V>.serialVersionUID"
       justification: "Serialization is not be used"
+    - code: "java.method.visibilityReduced"
+      old: "method org.apache.iceberg.view.ImmutableSQLViewRepresentation org.apache.iceberg.view.ImmutableSQLViewRepresentation::copyOf(org.apache.iceberg.view.SQLViewRepresentation)"
+      new: "method org.apache.iceberg.view.ImmutableSQLViewRepresentation org.apache.iceberg.view.ImmutableSQLViewRepresentation::copyOf(org.apache.iceberg.view.BaseSQLViewRepresentation)"
+      justification: "Immutables generated copyOf visibility changed"
```

Comment:
I think there must have been some behavior change in Immutables between the original code generation for SQLViewRepresentation and now. Immutables now generates a package-private copyOf (confirmed with ViewVersion). However, prior to this change the generated copyOf for SQLViewRepresentation was actually public. This is technically breaking to downgrade the copyOf but I don't see any options in the `Immutables` doc to control this behavior. Before merging this though, I'll explore a few options

Reply:
unfortunately it looks like the visibility of `copyOf()` is tied directly to the visibility of `BaseSQLViewRepresentation`

Reply:
You're right, it's tied to the visibility of the actual class which makes sense. Since we probably want to just go for 100% backwards compatibility I went with maintaining a copy of the Immutables generated class. I explored the option of generating a package private immutable class which we extend but Inheritance of the class isn't possible since it's final (Immutables only generates final classes as far as I can tell)

Reply:
I'll try to exlore some other alternatives this week

Reply:
@amogh-jahagirdar I've created https://github.com/apache/iceberg/pull/9302 which doesn't break the API and also doesn't require to copy anything from the generated class